fuzzing/test_case_assertion_provider.go

```diff
@@ -73,7 +73,44 @@ func (t *AssertionTestCaseProvider) checkAssertionFailures(callSequence calls.Ca
 	// have a panic code.
 	lastExecutionResult := lastCall.ChainReference.MessageResults().ExecutionResult
 	panicCode := abiutils.GetSolidityPanicCode(lastExecutionResult.Err, lastExecutionResult.ReturnData, true)
-	encounteredAssertionFailure := panicCode != nil && panicCode.Uint64() == abiutils.PanicCodeAssertFailed
+	encounteredAssertionFailure := false
+	if panicCode != nil {
+		// Define a map of config options and their corresponding panic codes
+		var panicCodes = map[string]int64{
+			"fail_on_assert_failed":                      abiutils.PanicCodeAssertFailed,
+			"fail_arithmetic_overflow":                   abiutils.PanicCodeArithmeticUnderOverflow,
+			"fail_divide_by_zero":                        abiutils.PanicCodeDivideByZero,
+			"fail_on_enum_type_conversion_out_of_bounds": abiutils.PanicCodeEnumTypeConversionOutOfBounds,
+			"fail_on_incorrect_storage_access":           abiutils.PanicCodeIncorrectStorageAccess,
+			"fail_on_pop_empty_array":                    abiutils.PanicCodePopEmptyArray,
+			"fail_on_out_of_bounds_array_access":         abiutils.PanicCodeOutOfBoundsArrayAccess,
+			"fail_on_allocate_too_much_memory":           abiutils.PanicCodeAllocateTooMuchMemory,
+			"fail_on_call_uninitialized_variable":        abiutils.PanicCodeCallUninitializedVariable,
+		}
+
+		// Get the value and type of the config options
+		configValue := reflect.ValueOf(t.fuzzer.config.Fuzzing.Testing.AssertionTesting.AssertionModes)
+		configType := configValue.Type()
+
+		// Iterate over each config option
+		for i := 0; i < configValue.NumField(); i++ {
+			// Get the value, type, and name of the current field
+			fieldValue := configValue.Field(i)
+			fieldType := configType.Field(i)
+			fieldName := fieldType.Tag.Get("json")
+
+			// Check if the config option is true
+			if fieldValue.Bool() {
+				// Get the panic code associated with the config option
+				code, found := panicCodes[fieldName]
+
+				// Check if the panic code matches the given panic code
+				if found && code == panicCode.Int64() {
+					encounteredAssertionFailure = true
+				}
+			}
+		}
+	}
```

Comment:
I think this overcomplicates the situation. Ideally, we maintain what panic modes are enabled within `AssertionTestCaseProvider`

Reply:
Even simpler would be to have a switch statement here like Xenomega mentioned. Will write the pseudocode

Reply:
Here is some pseudo-code that you can add to `attachAssertionTestCaseProvider`

```
if config.FailOnAssertion {
    t.panicCodes[abiutils.PanicCodeAssertFailed] = true
}
if config.FailOnUnderflow {
   t.panicCodes[abiutils.PanicCodeUnderflow] = true
}
```

Now in `checkAssertionFailure` you just have to do a key-value lookup in `AssertionTestCaseProvider.panicCodes`.  So `if t.panicCodes[panicCode] == true` then we can set `encounteredAssertionFailure` to true.